### ISSUE TITLE
buildModel Error, when Instance from class Annotion , example down descriptions

### ISSUE THREAD

**Issue by Scemoon** [INRIA/spoon#3510]
@Slf4j
public class Example {
   public void example(){
        log.info("exceptions")
   }
}
#--------------------------------
log instance not new or not import, from @Slf4j, so compile mode Exception

**Comment by MartinWitt** [comment]
Could you explain a bit more, what you are doing and which version you are using?
A testcase showcasing the problem is super helpful

**Comment by Scemoon** [comment]
Version 8.2.0, I use MavenLancher, when stage of buildModel , throw Not Found "log"  Exception. And log instance from @Slf4j  .

**Comment by MartinWitt** [comment]
Sry for the late response, im currently working on some older issues. But I still don't have any idea, what you are doing and what the error is? You have a maven project with only the source code given above and maven fails somehow magically? 

@monperrus can we introduce a "More Information needed" label and/or a bug report template? It' really hard to reproduce a bug without instructions or a testcase.

**Comment by monperrus** [comment]
Done, new label created: https://github.com/INRIA/spoon/labels/information-needed

Do you have the rights to label issue?

**Comment by MartinWitt** [comment]
No i don't have permission currently.

**Comment by Scemoon** [comment]
@MartinWitt detail debug for you .

![image](https://user-images.githubusercontent.com/4270790/133747194-5c5d74f5-1663-4d18-8341-a0a351223ddc.png)

**Comment by slarse** [comment]
This looks like a noclasspath issue to me, where Spoon isn't given enough information to determine if `log` is a method or a class. Are you running with or without providing a classpath?

Can you show the code you are analyzing?

And what version of Spoon are you using now? Can you try with 9.1.0? I somewhat recall that we made some improvements in this area over the past year.

**Comment by Scemoon** [comment]
> This looks like a noclasspath issue to me, where Spoon isn't given enough information to determine if `log` is a method or a class.
> 
> Can you show the code you are analyzing?
> 
> And what version of Spoon are you using now? Can you try with 9.1.0? I somewhat recall that we made some improvements in this area over the past year.

My code like this, use lombok Annotation. When Runing lombok can init log method. 
@slf4j
public class Example {
public void example(){
log.info("exceptions")
}
}

**Comment by slarse** [comment]
Lombok injecting stuff is a problem, because Spoon cannot know what it will inject. Here, Spoon cannot find what `log` is (because based on the source code alone, that's not possible), so it assumes that it is a class in `java.lang`.

There's a discussion about using Lombok and Spoon over at #3821, and potentially using `spoon-decompiler` on compiled class files to get the source code as rewritten by Lombok. I'm afraid there is no easy way to solve this, Spoon and Lombok are somewhat overlapping in functionality and conflicting in execution.

**Comment by Scemoon** [comment]
> Lombok injecting stuff is a problem, because Spoon cannot know what it will inject. Here, Spoon cannot find what `log` is (because based on the source code alone, that's not possible), so it assumes that it is a class in `java.lang`.
> 
> There's a discussion about using Lombok and Spoon over at #3821, and potentially using `spoon-decompiler` on compiled class files to get the source code as rewritten by Lombok. I'm afraid there is no easy way to solve this, Spoon and Lombok are somewhat overlapping in functionality and conflicting in execution.

Right.  I found conflict  with Lombok @data    when build mode with compiling，buildunits not found get or set method。
